bench/trace/hook/register.ts

```diff
@@ -28,6 +29,35 @@ experimentalAddHardhatNetworkMessageTraceHook(async ({ ethers }, trace) => {
   lastTrace = trace;
 });
 
+// NOTE: Leverage how hacky you can get with NodeJS and modify the tracer to
+// include additional gas information for instructions and subtraces. This code
+// should ideally be upstreamed and the experimental tracing extended to include
+// additional gas usage data.
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const VMTracerPrototype = VMTracer.prototype as any;
+const { _stepHandler, _afterMessageHandler } = VMTracerPrototype;
+VMTracerPrototype._stepHandler = function (step: any, next: any) {
+  const trace = this._messageTraces[this._messageTraces.length - 1] || {
+    steps: [],
+  };
```

Comment:
Does this mean that the gas from the first instruction is not accounted for? (Since `gasLeft` is attached to an object that only exists for this scope?)

Reply:
It can happen when transactions revert that a subtrace is created but no code is executed. This just makes sure we don't throw in that case.